core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -173,6 +172,7 @@ public synchronized void connect() throws IOException {
       if (mClosed) {
         throw new FailedPreconditionException("Failed to connect: client has been closed");
       }
+      mAddress = getAddress();
```

Comment:
NIT: Let's leave a comment like "get up-to-date address" or something.

Reply:
good idea, added